### PR TITLE
Созонов Илья. Задача 1. Вариант 7. Нахождение наиболее близких соседних элементов вектора.

### PR DESCRIPTION
**Pull request by sozozzya**
**Описание последовательной задачи:**
1. Создаём переменную min и инициализируем её максимальным возможным значением INT_MAX.
2. Проходим по вектору, начиная с первого элемента до предпоследнего, и находим модуль разности между текущим элементом и следующим за ним.
3. Если модуль разности между ними меньше текущего значения min, обновляем min и запоминаем индекс текущего элемента вектора.

**Описание MPI задачи:**
1. Создаем новый вектор diff, состоящий из пар: модуль разности соседних элементов исходного вектора и соответствующего индекса.
2. Полученный вектор пар делится равномерно на сегменты по числу процессов.
3. Каждый сегмент данных отправляется от нулевого процесса остальным.
4. Каждый процесс находит наименьший элемент и соответствующий ему индекс в полученном сегменте данных.
5. После завершения локальных вычислений процессы возвращают результаты корневому процессу, где с помощью коллективной операции reduce находится наименьший модуль разности соседних элементов вектора и соответствующий индекс.

### CODE REVIEW

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
+#include <gtest/gtest.h>
```
Comment (KolodkinGrigorii) [inline tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp:1]: add test for empty vector

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+#include <boost/mpi/communicator.hpp>
+#include <boost/mpi/environment.hpp>
+#include <vector>
+
+#include "mpi/sozonov_i_nearest_neighbor_elements/include/ops_mpi.hpp"
+
```
Comment (MatveyKurakin) [inline tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp:8]: Add a test on a small non-random vector that checks the correctness of the parallel and sequential algorithm

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp (hunk 1 above)
Comment (sozozzya) [reply tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp:1]: fixed, please check

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp (hunk 2 above)
Comment (sozozzya) [reply tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp:8]: fixed, please check

**Review by MatveyKurakin — APPROVED** [review]

**Review by KolodkinGrigorii — APPROVED** [review]

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/include/ops_mpi.hpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+
+namespace sozonov_i_nearest_neighbor_elements_mpi {
+
+std::vector<int> getRandomVector(int sz);
```
Comment (aobolensk) [inline tasks/mpi/sozonov_i_nearest_neighbor_elements/include/ops_mpi.hpp:17]: Please, move to the test module

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/include/ops_mpi.hpp (hunk 3 above)
Comment (sozozzya) [reply tasks/mpi/sozonov_i_nearest_neighbor_elements/include/ops_mpi.hpp]: fixed, please check

**Review by aobolensk — APPROVED** [review]

**Review by allnes — APPROVED** [review]
